crates/bevy_ecs/src/reflect/entity_commands.rs

```diff
@@ -0,0 +1,318 @@
+    ///         .remove_reflected(special_component_holder.component.clone_value(), type_registry.type_registry.clone());
+    /// }
+    ///
+    /// ```
+    fn remove_reflected(
+        &mut self,
+        component: Box<dyn Reflect>,
+        type_registry: TypeRegistry,
+    ) -> &mut Self;
+}
+
+impl<'w, 's, 'a> EntityCommandsReflectExtension for EntityCommands<'w, 's, 'a> {
+    fn insert_reflected(
+        &mut self,
+        component: Box<dyn Reflect>,
+        type_registry: TypeRegistry,
+    ) -> &mut Self {
+        self.commands.add(InsertReflected {
+            entity: self.entity,
+            type_registry,
+            component,
+        });
+        self
+    }
+
+    fn remove_reflected(
+        &mut self,
+        component: Box<dyn Reflect>,
+        type_registry: TypeRegistry,
+    ) -> &mut Self {
+        self.commands.add(RemoveReflected {
+            entity: self.entity,
+            type_registry,
+            component,
+        });
+        self
+    }
+}
+
+/// A [`Command`] that adds the boxed reflect component to an entity using the data in the provided
+/// [`TypeRegistry`].
+pub struct InsertReflected {
+    /// The entity on which the component will be inserted.
+    pub entity: Entity,
+    /// The [`TypeRegistry`] that the component is registered in and that will be used to get reflection
+    /// data in order to insert the component.
+    pub type_registry: TypeRegistry,
+    /// The reflect [`Component`] that will be added to the entity.
+    pub component: Box<dyn Reflect>,
+}
+
+impl Command for InsertReflected {
+    fn apply(self, world: &mut World) {
+        if let Some(mut entity) = world.get_entity_mut(self.entity) {
+            let type_info = self.component.type_name();
+            if let Some(type_registration) = self.type_registry.get_with_name(type_info) {
+                if let Some(reflect_component) = type_registration.data::<ReflectComponent>() {
+                    reflect_component.insert(&mut entity, &*self.component);
+                } else {
+                    panic!("Could not get ReflectComponent data (for component type {}) because it doesn't exist in this TypeRegistration.", self.component.type_name());
+                }
+            } else {
+                panic!("Could not get type registration for component (for component {}) because it doesn't exist in the TypeRegistry.", self.component.type_name());
+            }
+        } else {
+            panic!("error[B0003]: Could not insert a reflected component (of type {}) for entity {:?} because it doesn't exist in this World.", self.component.type_name(), self.entity);
+        }
+    }
+}
+
+/// A [`Command`] that removes the component of the same type as the given boxed reflect component from
+/// the provided entity. Does nothing if the entity does not have a component of the same
+/// type, if the [`TypeRegistry`] does not contain the data for the given component, or if the entity
+/// does not exist.
```

Comment:
Nit: This can just point back to the `EntityCommandsReflectExtension::remove_reflected` docs.
```suggestion
/// the provided entity.
///
/// See [`EntityCommandsReflectExtension::remove_reflected`] for details.
```